src/iceberg/transaction.h

```diff
@@ -39,6 +39,76 @@ class ICEBERG_EXPORT Transaction {
   /// \return this transaction's table
   virtual const std::shared_ptr<Table>& table() const = 0;
 
+  /// \brief Create a new schema addition operation
+  ///
+  /// \return a new AddSchema
+  virtual std::shared_ptr<AddSchema> AddSchema() = 0;
+
+  /// \brief Create a new set current schema operation
+  ///
+  /// \param schema_id the schema id to set as current
+  /// \return a new SetCurrentSchema
+  virtual std::shared_ptr<SetCurrentSchema> SetCurrentSchema(int32_t schema_id) = 0;
+
+  /// \brief Create a new remove schemas operation
+  ///
+  /// \param schema_ids the schema ids to remove
+  /// \return a new RemoveSchemas
+  virtual std::shared_ptr<RemoveSchemas> RemoveSchemas(
+      const std::vector<int32_t>& schema_ids) = 0;
```

Comment:
```suggestion
      std::span<const int32_t> schema_ids) = 0;
```

Should we use span to accept broader range of input types for here and below?